### PR TITLE
Use icon mixins and classes to add icons to elements

### PR DESCRIPTION
**Pull request by anitagraham**
- Rename  font to be more specific again (Refinery_icons)
- Simplify icon css 
- Modify Refinery css to use new font (some icons need to be added)
- Adjust  side-bar css to achieve same results as with icon markup
- Change some markup in the sidebar.

These are my notes on which icons may need to be added or changed
- plus   (using add-item, but plus, times, minus  preferred as they match size/width)  
- times  (called 'close', this describes function not the icon)
- minus  (using 'close' for delete, but minus preferred)
- warning (using 'close', but more specific icon preferred)
- download (using 'export'. This looks an export to cloud, rather than a download to local. Does it matter? I'm not sure)
- envelope (no suitable icon, using 'feather')
- info (no suitable icon, using 'feather')
- spinner (using 'gear')
- go (using 'arrow-right')
- question (using 'close')
- th (no suitable icon, using feather. th was used for image-grid-view)

I am using the times icon (X) icon for: close, delete, error, failure, warning

### CODE REVIEW

File: core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.site-name {@include icon('bars', $plugin_icon_colour, $margin: 1.2em)}
+
+.refinery_images_icon {@include icon('photo', $plugin_icon_colour, $margin: 1.2em)}
```
Comment (bricesanchez) [inline core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss]: At this time, i don't like line 3 to 6, icons should be defined in plugin config only : https://github.com/refinery/refinerycms/pull/3201/files#diff-24e9d953dbbc6362d29db46264449211L15

**Comment by bricesanchez** [comment]
Except my comment on icons config, it's ok for me.

What about @parndt and @stefanspicer ?

File: core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss (hunk 1 above)
Comment (anitagraham) [inline core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss]: There has to be some css _somewhere_ to link the class to the icon definition. We can change the class easily enough (from the first line below to the second), but it doesn't help us to get the icon on the page.  

```html
<a href="/refinery/pages" class="plugin bump-down refinery_pages_icon">Pages</a>
<a href="/refinery/pages" class="plugin bump-down files_icon">Pages</a>
```
You are right, it should be in the plugin, but be in `app/assets/plugin_name.css`, and add a `config/initializers/assets.rb` to add plugin_name.css to the asset pipeline. For example:

```scss
// refinerycms/pages/app/assets/refinery_pages.scss
.refinery_pages_icon {@include icon('files', $plugin_icon_colour, $margin: 1.2em)
```
----
An alternative is to create a class for each icon in the set, setting up a direct class<-->icon link.
The current class names are semantic classes, and the icon names are literal descriptions of the image.

```scss
.preview_icon {@include icon('eye', $icon_preview_colour)}
```
We could add 
```scss
.eye_icon {@include icon('eye')}
```
At that point we could use your original solution and write

```erb
  <%= link_to menu_item.title, refinery.url_for(menu_item.url), 
    class: ["plugin", "bump-down", "#{menu_item.icon}_icon"] << ("active" if menu_item.highlighted?(params))
  %>
```
to get

```html
<a href="/refinery/pages" class="plugin bump-down files_icon">Pages</a>
```

Comment (stefanspicer) [inline core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss]: Sorry Anita, I've been unable to do a thourough review of this PR today, but would it be helpful for us all to understand what problem we're trying to solve?

    <a href="/refinery/pages" class="plugin bump-down refinery_pages_icon">Pages</a>
    <a href="/refinery/pages" class="plugin bump-down files_icon">Pages</a>
    <a href="/refinery/pages"><i class="icon icon-files"></i> Pages</a>

What is the problem with the 3rd option there (the current html)?

Comment (stefanspicer) [inline core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss]: I remembered we already started this conversation, so I guess you've answered my question here: https://github.com/refinery/refinerycms/pull/3167#issuecomment-222154804

I just reviewed it.  Here's my summary:

1. The <i></i> tag is superfluous markup
1. The erb is cleaner
1. Easier to change icons
1. The markup and the code is cleaner
1. A single place to see which icons are being used and how to use them
1. Mixins offer consistency and flexibility when generating icons

@anitagraham are these your reasons?  @bricesanchez you think this PR solves all of these?

Comment (bricesanchez) [inline core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss]: @anitagraham My only concern is to not have a new CSS file in each extension to define an icon. I f not we will have to update the form/engine generator to add this new file.

I prefer the alternative to pre-create all class for each icon in core.

@stefanspicer it should be.

File: core/app/assets/stylesheets/glass/components/_icons.scss [hunk 2]
```diff
@@ -198,232 +198,230 @@
   color: #fff;
 }
 
-body {
-  .icon:not(.icon-fe) {
```
Comment (stefanspicer) [inline core/app/assets/stylesheets/glass/components/_icons.scss]: This `not(.icon-fe)` will be necessary in the future, but we can have a separate PR to pull that back in.

File: core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss (hunk 1 above)
Comment (anitagraham) [inline core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss]: First: although I'll argue against your points @bricesanchez, I'll actually do what you've suggested.

I think that adding a s/css file and modifying a generator is not a big problem either initially or in maintenance.  

Still trying to work out how to do it your way (within the plugin definition) and my way (without adding yet another set of classes.)

File: core/app/assets/stylesheets/glass/components/_icons.scss (hunk 2 above)
Comment (anitagraham) [inline core/app/assets/stylesheets/glass/components/_icons.scss]: I only removed it because it seemed to generate a lot of css.

**Comment by anitagraham** [comment]
Closing this while I integrate @stefanspicer 's fixes into this.

**Comment by stefanspicer** [comment]
@anitagraham you should just be able to commit my changes to your glass_icons branch and it will update this PR. No need for a new one.

**Comment by anitagraham** [comment]
I think that's what I ended up doing Stefan. But first I did it the long way round. Anyway your fixes and mine should now be in this PR.
Anyway, it looks good to the naked eye, and the checks passed.

**Comment by bricesanchez** [comment]
There are regressions on UI like the icon hover color in the menu and the info icon looks missing :
company_name_-_refinery_et_refinerycms_ __rails_command_server_-p_4000_ _ruby_ _158x23_et_gemfile_ _refinerycms
company_name_-_refinery_et_refinerycms_ __rails_command_server_-p_4000_ _ruby_ _158x23

**Comment by bricesanchez** [comment]
Should we merge this PR before https://github.com/refinery/refinerycms-authentication-devise/pull/23 ?

If yes, could you fix and rebase this branch @anitagraham?

**Comment by anitagraham** [comment]
Will do

**Comment by stefanspicer** [comment]
@anitagraham I sent you these files in https://github.com/anitagraham/refinerycms/pull/3/:

```
$ md5sum refinery-icons.*
91fc9c1f23904aed53eb21cdf659c413  refinery-icons.eot
0ca45d61cbf27344a846d8db9a930994  refinery-icons.otf
47d431a4e3dd0025fbf067e14cfa17c8  refinery-icons.ttf
df570fecbddadba2bfa3f9410f1682fd  refinery-icons.woff
```

You have these files in this PR:

```
$ md5sum refinery-icons.*
91fc9c1f23904aed53eb21cdf659c413  refinery-icons.eot
0ca45d61cbf27344a846d8db9a930994  refinery-icons.otf
47d431a4e3dd0025fbf067e14cfa17c8  refinery-icons.ttf
a8640f30850cb5a40731de8fa121c254  refinery-icons.woff
```

So it looks like you took my changes for the eot, otf, ttf, but rejected my woff changes?  Or have you added something else to the woff since?

**Comment by bricesanchez** [comment]
Any update @anitagraham ?

**Comment by bricesanchez** [comment]
I close this PR because we don't use glass admin ui for now.

**Pull request closed without merging**